plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/dialog/FileOpenSaveDialog.java

```diff
@@ -449,11 +451,174 @@ protected void createOpenLayout( Composite parent, Composite select ) {
+
+          // Skip the single "/" when accessing repository
+          if ( !children.isEmpty() && children.get( 0 ) instanceof RepositoryFile ) {
+            targetPathArrayIndex = 1;
+          } else {
+            targetPathArrayIndex = 0;
+          }
+
+
+          do {
+            currentFile = null;
+
+            if ( targetPathArrayIndex == targetPathArray.length ) {
+              break;
+            }
+            for ( File file : children ) {
+              if ( isFileEqual( file, targetPathArray[ targetPathArrayIndex ] ) ) {
+                currentFile = file;
+                break;
+              }
+            }
+            if ( currentFile instanceof Directory ) {
+              treeViewer.setSelection( new StructuredSelection( currentFile ), true );
+              treeViewer.setExpandedState( currentFile, true );
+              try {
+                children = FILE_CONTROLLER.getFiles( currentFile, null, true );
+                // Sort in increasing order
+                sortFileList( children );
+                targetPathArrayIndex++;
+              } catch ( FileException e ) {
+                // Ignore
+              }
+            }
+          } while ( currentFile != null );
+        }
+      }
+    }
+  }
+
+  private boolean isFileEqual( File file, String fileName ) {
+    boolean isFileEqual = false;
+    if ( file instanceof VFSFile ) {
+      if ( file instanceof VFSLocation ) {
+        String pathName = ( (VFSLocation) file ).getConnectionPath();
+        pathName = pathName.substring( 0, pathName.length() - 1 ); // Removes last "/" for comparison
+        isFileEqual = pathName.equals( fileName );
+      } else if ( ( (VFSFile) file ).getConnectionPath().equals( fileName ) ) {
+        isFileEqual = true;
+      }
+    } else if ( file.getPath().equals( fileName ) ) {
+      isFileEqual = true;
+    }
+    return isFileEqual;
+  }
+
+  private void sortFileList( List<File> children ) {
+    if ( children.get( 0 ) instanceof VFSFile ) {
+      children.sort(
+        ( f1, f2 ) -> ( (VFSFile) f2 ).getConnectionPath().length() - ( ( (VFSFile) f1 ).getConnectionPath()
+          .length() ) );
+    } else {
+      children.sort( ( f1, f2 ) -> ( f2 ).getPath().length() - ( ( f1 ).getPath().length() ) );
+    }
+  }
+
+  private String[] getStringsAtEachDirectory( String targetPath, char pathSplitter ) {
+    String[] targetPathArray;
+
+    int currentTargetPathArrayIndex = 0;
+    if ( pathSplitter == '\\' ) {
+      int arraySize = targetPath.split( "\\\\" ).length + 1;
+      targetPathArray = new String[ arraySize ];
+      for ( int i = 0; i < targetPath.length(); i++ ) {
+        if ( targetPath.charAt( i ) == pathSplitter || i == targetPath.length() - 1 ) {
+          targetPathArray[ currentTargetPathArrayIndex ] = targetPath.substring( 0, i + 1 );
+          currentTargetPathArrayIndex++;
+        }
+      }
+    } else {
+      // VFS File Path
+      if ( targetPath.contains( "//" ) ) {
```

Comment:
Probably best to look for pvfs:// or hc:// here, if the scheme is still included.  Let's be explicit.